### PR TITLE
[GPU] Added a new unittest to test the fused external primitive

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - When external id of a loop is fused, the i/o map should be updated too. Added a unittest to check that occasion.
 - This is a following up task of https://github.com/openvinotoolkit/openvino/pull/10726

### Tickets:
 - 80559

### CODE REVIEW

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+             input_primitive_maps, output_primitive_maps, back_edges, p.loop_trip_count),
+        reorder("output", "loop", format::bfyx, p.default_type)
+    );
+
+    tolerance = 1e-5f;
+    execute(p);
+}
+
+#define CASE_LOOP_F32_1 3, { 1, 8, 3, 2 }, { 1, 2, 8, 3 }, {1, 2, 8, 1}, { 0, 3, 1, 2 }, data_types::f32, format::bfyx, data_types::f32
+#define CASE_LOOP_F16_0 4, { 1, 12, 4, 2 }, { 1, 2, 12, 4 }, {1, 2, 12, 1}, { 0, 3, 1, 2 }, data_types::f16, format::bfyx, data_types::f16
```
Comment (lznamens) [inline src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp:93]:
```suggestion
#define CASE_LOOP_F32_1 3, { 1, 8, 3, 2 }, { 1, 2, 8, 3 }, { 1, 2, 8, 1 }, { 0, 3, 1, 2 }, data_types::f32, format::bfyx, data_types::f32
#define CASE_LOOP_F16_0 4, { 1, 12, 4, 2 }, { 1, 2, 12, 4 }, { 1, 2, 12, 1 }, { 0, 3, 1, 2 }, data_types::f16, format::bfyx, data_types::f16
```

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2018-2022 Intel Corporation
```
Comment (lznamens) [inline src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp:1]: I suppose it's a new file => 2022

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    set_values(num_iteration_mem, {0});
+    set_values(trip_count_mem, {p.loop_trip_count});
+    set_values(initial_condition_mem, {1});
+    topology body(
+        input_layout("body_input", layout{p.data_type, format::bfyx, p.loop_eltwise_shape}),
+        input_layout("body_eltwise_operand", layout({p.data_type, format::bfyx, p.loop_eltwise_shape})),
+        eltwise("body_eltwise", "body_input", "body_eltwise_operand", eltwise_mode::sum)
+    );
+
+    std::vector<loop::io_primitive_map> input_primitive_maps {loop::io_primitive_map("eltwise", "body_input", 2), loop::io_primitive_map("loop_eltiwse_init_values", "body_eltwise_operand")};
```
Comment (lznamens) [inline src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp:69]: Please split onto 2 separate lines.

Comment (yeonbok) [reply src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp]: Fixed, thanks

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp]: Fixed, thanks

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                                                              loop::io_primitive_map("loop_eltiwse_init_values", "body_eltwise_operand")};
+    std::vector<loop::io_primitive_map> output_primitive_maps {loop::io_primitive_map("loop", "body_eltwise", 2)};
+    std::vector<loop::backedge_mapping> back_edges {loop::backedge_mapping("body_eltwise", "body_eltwise_operand")};
+
+    create_topologies(
+        input_layout("input", get_input_layout(p)),
+        data("eltwise_data", get_mem(layout{p.data_type, p.default_format, p.loop_input_shape})),
+        permute("permute", "input", p.permute_order),
+        eltwise("eltwise", {"permute", "eltwise_data"}, eltwise_mode::sum, p.data_type),
+        data("loop_eltiwse_init_values", get_mem(layout{p.data_type, format::bfyx, p.loop_eltwise_shape}, 0.f)),
```
Comment (lznamens) [inline src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp:79]:
```suggestion
        data("loop_eltwise_init_values", get_mem(layout{p.data_type, format::bfyx, p.loop_eltwise_shape}, 0.f)),
```

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        data("trip_count", trip_count_mem),
+        data("initial_condition", initial_condition_mem),
+        mutable_data("num_iteration", num_iteration_mem),
+        loop("loop", {"eltwise", "loop_eltiwse_init_values"}, body,
```
Comment (lznamens) [inline src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp:83]:
```suggestion
        loop("loop", {"eltwise", "loop_eltwise_init_values"}, body,
```

**Review by lznamens — APPROVED** [review]
Thanks, LGTM. I found another typo, sorry that didn't see it in a first time.

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp (hunk 4 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp]: Thanks ! I fixed

File: src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp (hunk 5 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/tests/fusions/loop_fusion_test.cpp]: Fixed

**Pull request merged**